src/main/java/com/epam/aidial/cfg/web/security/MultiPathGrantedAuthoritiesConverter.java

```diff
@@ -0,0 +1,78 @@
+package com.epam.aidial.cfg.web.security;
+
+import lombok.Data;
+import lombok.extern.slf4j.Slf4j;
+import org.jetbrains.annotations.NotNull;
+import org.springframework.core.convert.converter.Converter;
+import org.springframework.security.core.GrantedAuthority;
+import org.springframework.security.core.authority.SimpleGrantedAuthority;
+import org.springframework.security.oauth2.jwt.Jwt;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+@Data
+@Slf4j
+public class MultiPathGrantedAuthoritiesConverter implements Converter<Jwt, Collection<GrantedAuthority>> {
+
+    private static final String DEFAULT_AUTHORITY_PREFIX = "SCOPE_";
+    private String authorityPrefix = DEFAULT_AUTHORITY_PREFIX;
+
+    private List<String> authoritiesPaths;
+
+    @NotNull
+    @Override
+    public List<GrantedAuthority> convert(@NotNull Jwt token) {
+        Collection<GrantedAuthority> grantedAuthorities = new ArrayList<>();
```

Comment:
Can we double check that we really need `grantedAuthorities` here?  It looks like it isn’t used anywhere below.